### PR TITLE
Feature/remove user

### PR DESCRIPTION
**Pull request by aroon0808**
Changed searchbar to rerender users list after single letter press.
Changed relations has many belongs to to has many through with intermediary model workers and added to it archived field.
Now Admin can delete user from project.

### CODE REVIEW

File: app/javascript/packs/list.jsx [hunk 1]
```diff
@@ -38,21 +40,37 @@ class List extends React.Component {
         return -1;
       }
     });
+
     const projects = this.state.projects;
     const projectsList = projects.map(project => {
+      const ProjectUser = project.users.map(user => {
+        console.log(project.workers);
```
Comment (Vakrim) [inline app/javascript/packs/list.jsx]: remove console log

File: app/javascript/packs/list.jsx [hunk 2]
```diff
@@ -38,21 +40,37 @@ class List extends React.Component {
...
+          <div className="padding">
+            <span>{user.full_name}</span>{" "}
+            <DeleteUser UserId={user.id} ProjectId={project.id}>
+              Delete
+            </DeleteUser>
+          </div>
+        );
+      });
+
+      const ProjectId = project.workers.map(e => {
```
Comment (Vakrim) [inline app/javascript/packs/list.jsx]: unused constant?

File: app/javascript/packs/list.jsx [hunk 3]
```diff
@@ -38,21 +40,37 @@ class List extends React.Component {
         return -1;
       }
     });
+
     const projects = this.state.projects;
     const projectsList = projects.map(project => {
+      const ProjectUser = project.users.map(user => {
```
Comment (Vakrim) [inline app/javascript/packs/list.jsx]: `numberList` is now very long and has multiple responsibilities. You should split it

File: app/views/layouts/logged.html.erb [hunk 4]
```diff
@@ -2,13 +2,11 @@
 <html>
   <head>
     <title>Pik</title>
-    <%= csrf_meta_tags %>
-    <%= csp_meta_tag %>
-
-    <%= stylesheet_link_tag    'application', media: 'all' %>
-    <%= javascript_include_tag 'application' %>
+    <%= csrf_meta_tags %> <%= csp_meta_tag %> <%=
```
Comment (Vakrim) [inline app/views/layouts/logged.html.erb]: why this change?

File: app/controllers/api/v1/projects_controller.rb [hunk 5]
```diff
@@ -11,6 +11,11 @@ def not_assigned_users
     render json: @not_assigned_users
   end
 
+  def users_in_project
+    project_users = Project.find(params[:id]).users
+    render json: project_users
```
Comment (hirokiraj) [inline app/controllers/api/v1/projects_controller.rb]: use jbuilder

File: app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -37,6 +36,12 @@ def archive
     head :no_content
   end
 
+  def archive_user
+    worker = @projects.workers.find_by(user_id: params[:user_id])
```
Comment (hirokiraj) [inline app/controllers/projects_controller.rb]: why `@projects`? If we can call `workers` on it then this is single instance of the model, right?

File: app/controllers/projects_controller.rb [hunk 7]
```diff
@@ -37,6 +36,12 @@ def archive
...
+    worker.update(archived: true)
+    head :no_content
```
Comment (hirokiraj) [inline app/controllers/projects_controller.rb]: do return 204 if there is still object existing underneath

File: config/routes.rb [hunk 8]
```diff
@@ -25,6 +25,7 @@
 
   root to: 'welcome#index'
   patch '/projects/archive/:id' => 'projects#archive'
+  patch '/projects/:id/archive_user/:user_id' => 'projects#archive_user'
```
Comment (hirokiraj) [inline config/routes.rb]: why this is not in `api/v1` namespace? It returns just `head` and looks like api endpoint

**Comment by hirokiraj** [comment]
a file `report.20190729.162535.15354.0.001.json`, what is it doing?

**Comment by hirokiraj** [comment]
provide tests for relations and controller actions
